### PR TITLE
[CDAP-7734] Fix broken CLI

### PR DESCRIPTION
**Pull request by ecfm**
https://issues.cask.co/browse/CDAP-7734
http://builds.cask.co/browse/CDAP-DUT5170-2

### CODE REVIEW

**Comment by gokulavasan** [comment]
@maochf @bdmogal Had a question about unit test catching this issue in the JIRA. Was there a unit test that was lacking? If yes, can we add one?

File: cdap-cli/src/main/java/co/cask/cdap/cli/completer/supplier/EndpointSupplier.java [hunk 1]
```diff
@@ -46,9 +49,10 @@ private EndpointSupplier(final ServiceClient serviceClient, CLIConfig cliConfig)
   public Completer getCompleter(String prefix, Completer completer) {
     if (prefix != null && !prefix.isEmpty()) {
       String prefixMatch = prefix.replaceAll("<.+?>", "<>");
-      if (METHOD_PREFIX.equals(prefixMatch)) {
+      if ((METHOD_PREFIX.equals(prefixMatch) || METHOD_PREFIX_WITH_APP_VERSION.equals(prefixMatch))
```
Comment (albertshau) [inline cdap-cli/src/main/java/co/cask/cdap/cli/completer/supplier/EndpointSupplier.java:56]: could you add some comments here about what command prefixes these conditions handle?

**Review by albertshau — COMMENTED** [review]
would it be possible to add a unit test for this?

**Comment by ecfm** [comment]
@gokulavasan Currently there's no easy way to test this. Opened a JIRA: https://issues.cask.co/browse/CDAP-7760

**Comment by chtyim** [comment]
Can we add a test that just call `CLIMain.main(new String[0])`? You might need to call it from a different thread and issue interrupt, or try to send a exit command to it after it starts.

**Comment by albertshau** [comment]
changes lgtm pending investigation on what can be done for unit test

**Comment by chtyim** [comment]
The test LGTM

**Review by albertshau — APPROVED** [review]
LGTM let's merge!

File: cdap-cli-tests/src/test/java/co/cask/cdap/cli/CLIMainTest.java [hunk 2]
```diff
@@ -168,6 +175,44 @@ public void testConnect() throws Exception {
...
+        public Boolean call() throws Exception {
+          return cliPromptPattern.matcher(outputStream.toString()).find();
+        }
+      }, 20, TimeUnit.SECONDS, 200, TimeUnit.MILLISECONDS);
+
+      future.cancel(true);
+    } catch (Exception e) {
+      LOG.error(e.getMessage());
+      Assert.fail("Fail to find CLI prompt from the output of CLIMain.main");
+      if (future != null) {
```
Comment (anwar6953) [inline cdap-cli-tests/src/test/java/co/cask/cdap/cli/CLIMainTest.java]: code will never get to this part, right? the previous line will guarantee that.

File: cdap-cli-tests/src/test/java/co/cask/cdap/cli/CLIMainTest.java [hunk 3]
```diff
@@ -168,6 +175,44 @@ public void testConnect() throws Exception {
   }
 
   @Test
+  public void testMain() {
+    final InputStream systemIn = System.in;
+    final PrintStream systemOut = System.out;
+    final ByteArrayOutputStream outputStream = new ByteArrayOutputStream();
+    final PrintStream printStream = new PrintStream(outputStream);
+    final Pattern cliPromptPattern = Pattern.compile("cdap \\(.+\\)>");
+    ExecutorService executorService = Executors.newFixedThreadPool(1);
```
Comment (anwar6953) [inline cdap-cli-tests/src/test/java/co/cask/cdap/cli/CLIMainTest.java]: there's already the method `newSingleThreadExecutor`, right?

File: cdap-cli-tests/src/test/java/co/cask/cdap/cli/CLIMainTest.java [hunk 4]
```diff
@@ -168,6 +175,44 @@ public void testConnect() throws Exception {
   }
 
   @Test
+  public void testMain() {
+    final InputStream systemIn = System.in;
```
Comment (anwar6953) [inline cdap-cli-tests/src/test/java/co/cask/cdap/cli/CLIMainTest.java]: seems unnecessary to keep track of systemIn?

**Comment by anwar6953** [comment]
lgtm

**Comment by ecfm** [comment]
Temporarily close to debug and fix build failure

**Comment by ecfm** [comment]
Remove the test and will open separate PR to add the test

**Review by albertshau — APPROVED** [review]
LGTM. Test was facing complications, so punting on it for now. Don't think it is terribly important since this type of error should be easily caught in the future as long as somebody actually tries using the CLI.

**Pull request merged**